larky/src/main/resources/vendor/iso8583/encoder.star

```diff
@@ -399,16 +399,21 @@ def _encode_field(doc_dec, doc_enc, field_key, spec):
         # BCD LLVAR length \x99 must be string "99"
         # BCD LLLVAR length \x09\x99 must be string "0999"
         # BCD LLLLVAR length \x99\x99 must be string "9999"
+        # "{:0{len_type}d}".format(enc_field_len, len_type=len_type * 2)
+        enc_field_len_str = str(enc_field_len)
```

Comment:
Is this a bug in the actual upstream library code?

Reply:
nope, upstream library code is correct, but it seems that string formatting wasn't translated properly into larky (original code is commented on line 402, btw)

e.g., for BCD case if `enc_field_len = 5` and `len_type = 2`, original code will result into:
`{:0{len_type}d}".format(enc_field_len, len_type=len_type * 2)` => `"0005"`

but in larky code we had:
`"{enc_field_len}".format(enc_field_len=enc_field_len)` => `5`

Reply:
`"{enc_field_len}".format(enc_field_len=enc_field_len)` is equivalent to just `str(enc_field_len)`, btw

Reply:
Ah I see, the padding was the issue. Ok. That makes sense